### PR TITLE
luxon: Fix DateTimeFormatOptions to be compatible with ES2020.Intl

### LINKED ISSUE #50392: [@types/luxon] ts 4.2 - DateTimeFormatOptions conflict
@colbydehart @FourwingsY @jsiebern @mastermatt @pietrovismara @dawnmist @ycmjason @Aitor1995 @peterblazejewicz

I'm not sure what is wrong, but I think `es2020.intl` is conflicting with current types. Example code bellow, but even tests are failing with `TypeScript 4.2`
```ts
dt.toLocaleString(DateTime.DATETIME_SHORT_WITH_SECONDS);
```
```txt
Argument of type 'DateTimeFormatOptions' is not assignable to parameter of type 'LocaleOptions & DateTimeFormatOptions'.
  Type 'DateTimeFormatOptions' is not assignable to type 'LocaleOptions'.
    Types of property 'numberingSystem' are incompatible.
      Type '"arab" | "arabext" | "bali" | "beng" | "deva" | "fullwide" | "gujr" | "guru" | "hanidec" | "khmr" | "knda" | "laoo" | "latn" | "limb" | "mlym" | "mong" | "mymr" | "orya" | "tamldec" | ... 69 more ... | undefined' is not assignable to type '"arab" | "arabext" | "bali" | "beng" | "deva" | "fullwide" | "gujr" | "guru" | "hanidec" | "khmr" | "knda" | "laoo" | "latn" | "limb" | "mlym" | "mong" | "mymr" | "orya" | "tamldec" | "telu" | "thai" | "tibt" | undefined'.
        Type '"adlm"' is not assignable to type '"arab" | "arabext" | "bali" | "beng" | "deva" | "fullwide" | "gujr" | "guru" | "hanidec" | "khmr" | "knda" | "laoo" | "latn" | "limb" | "mlym" | "mong" | "mymr" | "orya" | "tamldec" | "telu" | "thai" | "tibt" | undefined'.
```

Test: 
```txt
$ dtslint types luxon
Error: C:/[...]/DefinitelyTyped/types/luxon/luxon-tests.ts:133:35
ERROR: 133:35  expect  TypeScript@4.2 compile error:
Argument of type '{ month: string; day: string; }' is not assignable to parameter of type 'LocaleOptions & DateTimeFormatOptions'.
  Type '{ month: string; day: string; }' is not assignable to type 'DateTimeFormatOptions'.
    Types of property 'month' are incompatible.
      Type 'string' is not assignable to type '"long" | "short" | "narrow" | "numeric" | "2-digit" | undefined'.
ERROR: 134:38  expect  TypeScript@4.2 compile error:
Argument of type '{ month: string; day: string; }' is not assignable to parameter of type 'LocaleOptions & DateTimeFormatOptions'.
  Type '{ month: string; day: string; }' is not assignable to type 'DateTimeFormatOptions'.
ERROR: 135:38  expect  TypeScript@4.2 compile error:
Argument of type '{ month: string; day: string; }' is not assignable to parameter of type 'LocaleOptions & DateTimeFormatOptions'.
  Type '{ month: string; day: string; }' is not assignable to type 'DateTimeFormatOptions'.

    at C:\[...]\DefinitelyTyped\node_modules\dtslint\bin\index.js:206:19
    at Generator.next (<anonymous>)
    at fulfilled (C:\[...]\DefinitelyTyped\node_modules\dtslint\bin\index.js:6:58)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

### PR DESCRIPTION
**Pull request by KapitanOczywisty**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.  -- *Adding `ES2020.Intl` to `tsconfig.json` causes tests to fail on ts4.2 with current definitions.*
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 

TL;DR [This definition](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/187d62800d9e1f033841d022e37f44e3cff358a9/types/luxon/index.d.ts#L379-L417) is not compatible with [ES2020.Intl](https://github.com/microsoft/TypeScript/blob/290b0ea94d540bb451880cc598a7f1fbf93c848e/src/lib/es2020.intl.d.ts#L64-L288)

You can see change on [Playground](https://www.typescriptlang.org/play?target=7&module=0&ts=4.2.0-beta#code/JYWwDg9gTgLgBAbzgEQIYwKYBVQbgXzgDMoIQ4ByACxhjAGcAuAemalQHcA6Ac2BioBXAEaD6GKAGMIAO0xyu0kM2QYiwGfwwAbAJ5ZdYDABMVajVr0GjpgIwAOAOzGAbACZ7ABk-GAnBlsiTwBmYPsAFltjTzc3DGDHInDw+MkiImCAVntUX2YYQwx6Zm1BAA9ZZg1jDDKuYy4YegoAbgBYAChQSFhEFHRsXDhUen7MHBAMADFgMoJiUnJqWgYWNk5efiFRcSlZeRhFMmYAaVQwflQZAHlJAC9dDmB6ArN1TUwrQtNSipkAWia4S4bnyhWKv0q1Vq9UazXaHU6CE6cFRcGkMhecGM8AAvGNBpMuNoIJJUNoABQASgRKLROMaEAAMqTyRgAMowKAaHgUtDjXBcZAAQSwAFEsABJACyYoA+uyABLXABKWDlAHVJVhFQqxQBha4AOWQ7JpnXwSLpqIxWJxcHx-MJ01mxNZlPNiI6aOxhxgzPdHK5PL5AwmLrqIvFUtlCuVas12t17INxtNnvwQA)

Fixes #50392

### CODE REVIEW

File: types/luxon/tsconfig.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
     "compilerOptions": {
         "module": "commonjs",
         "lib": [
-            "es6"
+            "es6", "ES2020.Intl"
```
Comment (peterblazejewicz) [inline types/luxon/tsconfig.json:5]: is this correct?
https://github.com/SchemaStore/schemastore/blob/master/src/schemas/json/tsconfig.json#L553-L557
(formally, for devex, it does not pass validation in editors)

Comment (KapitanOczywisty) [reply types/luxon/tsconfig.json:5]: @peterblazejewicz  It's present in ts for quite some time, but only ts4.2 version is causing problems. FWIW `es2020` itself gives the same results, so we can replace for that.

Comment (peterblazejewicz) [reply types/luxon/tsconfig.json:5]: yup, let's swap, thx!

Comment (KapitanOczywisty) [reply types/luxon/tsconfig.json]: @peterblazejewicz fixed, let me know if there is something else

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@KapitanOczywisty thx!

**Review by rbuckton — APPROVED** [review]

**Comment by KapitanOczywisty** [comment]
Ready to merge 😃
